lib/api/wpc.h

```diff
@@ -944,4 +954,182 @@ app_res_e WPC_register_for_stack_status(onStackStatusReceived_cb_f onStackStatus
+
+/**
+ *  \brief Reservation for maximum test data size in radio interface
+ *
+ *  Test data header includes length (1 Byte) and sequence
+ *  number (4 Bytes) which allocates fixed 5 bytes from over the air
+ *  payload, remaining bytes can be used for free form data.
+ *
+ *  Maximum data size depends on used radio hardware. Here we allocate
+ *  big enough buffer to cover all supported radio hardwares and
+ *  some extra bytes for future needs.
+ *
+ *  True size of the radio payload can be read with test_lib->getMaxDataSize().
+ */
+#define APP_TEST_MODE_MAX_DATA_SIZE (180 - 1 - 4)
+
+/**
+ * \brief Header of test data send over the air
+ */
+typedef struct
+{
+    /** Sequence number of the message  */
+    uint32_t seq;
+    /** Length of the test data  */
+    uint8_t len;
+} app_test_mode_data_header_t;
+
+/**
+ * \brief   Structures for test data reception
+ */
+typedef struct
+{
+    /** Number of received messages since test mode started */
+    uint32_t cntr;
+    /** Number of duplicate messages since test mode started
+     *  Incremented if received message has the same sequence number as
+     *  in previous test data package */
+    uint32_t dup;
+} app_test_mode_data_counters_t;
+
+typedef struct
+{
+    /** Received signal strength in dBm with resolution of 1dBm. */
+    int8_t rssi;
+    /** Received data counters. */
+    app_test_mode_data_counters_t rxCntrs;
+    /** Header of received test data */
+    app_test_mode_data_header_t hdr;
+    /** Received test data */
+    uint8_t data[APP_TEST_MODE_MAX_DATA_SIZE];
+} app_test_mode_data_received_t;
+
+/**
+ * \brief   Structures for test data transmission
+ */
+typedef struct
+{
+    /** Number of messages to be sent per request.
+        Bigger the bursts value is the longer the transmission takes and delays
+        response to the caller.  Practical values are from 1 to 1000. */
+    uint32_t bursts;
+    /** Time for Clear Channel Assessment in us. If 0, no CCA done.
+        Practical values are from 0 to tens of milliseconds */
+    uint32_t ccaDuration;
+    /** When CCA is not used (ccaDuration=0) transmit interval defines the delay between
+     *  TX bursts in us. If 0, sending done as soon as possible. */
+    uint32_t txInterval;
+} app_test_mode_data_transmitter_control_t;
+
+typedef struct
+{
+    /** Header for fixed data fields */
+    app_test_mode_data_header_t hdr;
+    /** Free form test data */
+    const uint8_t * data;
+} app_test_mode_data_tx_payload_t;
+
+typedef struct
+{
+    /** Parameters for test data transmission */
+    app_test_mode_data_transmitter_control_t txCtrl;
+    /** Data to be transmitted over the air */
+    app_test_mode_data_tx_payload_t txPayload;
+} app_test_mode_data_transmit_t;
+
+typedef struct
+{
+    /** Parameters for test data transmission */
+    app_test_mode_data_transmitter_control_t txCtrl;
+    /** Test signal type to be trasmitted */
+    //app_test_mode_signal_type_e  signalType;
```

Comment:
Is it needed? Why is it commented?

Reply:
It would be a cleaner way to have own enum instead of uint8_t in wpc.h for this. I made an reservation for it but I forgot to add it while I was in hurry to test this so I did it now.   Will be fixed in next commit.